### PR TITLE
`azurerm_monitor_diagnostic_setting` - fix `log_analytics_destination_type` 

### LINKED ISSUE #20140: `azurerm_monitor_diagnostic_setting` allways want to change `log_analytics_destination_type` from empty to `AzureDiagnostics` starting with 3.40.0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.40.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_diagnostic_setting

### Terraform Configuration Files

```hcl
resource "azurerm_monitor_diagnostic_setting" "subscription" {
  name                       = "Subscription"
  target_resource_id         = data.azurerm_subscription.current.id
  log_analytics_workspace_id = azurerm_log_analytics_workspace.log_analytics.id

  enabled_log {
    category = "Administrative"
  }
  enabled_log {
    category = "Alert"
  }
  enabled_log {
    category = "Recommendation"
  }
  enabled_log {
    category = "ResourceHealth"
  }
  enabled_log {
    category = "Security"
  }
  enabled_log {
    category = "ServiceHealth"
  }
}
```

### Debug Output/Panic Output

```shell
{"@level":"info","@message":"azurerm_monitor_diagnostic_setting.subscription: Plan to update","@module":"terraform.ui","@timestamp":"2023-01-20T17:05:28.962335Z","change":{"resource":{"addr":"azurerm_monitor_diagnostic_setting.subscription","module":"","resource":"azurerm_monitor_diagnostic_setting.subscription","implied_provider":"azurerm","resource_type":"azurerm_monitor_diagnostic_setting","resource_name":"subscription","resource_key":null},"action":"update"},"type":"planned_change"}
```

### Expected Behaviour

No changes to the `azurerm_monitor_diagnostic_setting` is planed

### Actual Behaviour

In every plan `azurerm_monitor_diagnostic_setting` wants to set `log_analytics_destination_type`again to `AzureDiagnostics`.
grafik

### Steps to Reproduce

_No response_

### Important Factoids

This starts with 3.40.0. Is working with 3.39.1

### References

_No response_

### PR DESCRIPTION
**Pull request by teowa**
resolves #20140

The issue is caused by service  API behavior change and related to #20048: 
- **Before**: some resources had `log_analytics_destination_type` property default to `AzureDiagnostics`. So PR #20048 was introduced to set TF schema property default to `AzureDiagnostics`.
- **Now**:  some resources (e.g. `azurerm_key_vault`) returns empty for `log_analytics_destination_type` property even if you set it as `AzureDiagnostics` or `Dedicated`  in request. But some other resources (e.g. `azurerm_data_factory`) still  default to `AzureDiagnostics`. Details is in https://github.com/Azure/azure-rest-api-specs/issues/22400.

The above API behavior change happened after PR https://github.com/hashicorp/terraform-provider-azurerm/pull/20048 (set `log_analytics_destination_type` default to `AzureDiagnostics`) was merged. Days ago the API also returns `AzureDiagnostics` for resource like Key Vault, this can be proved by daily test record, it works well before Jan 26, and PR #20048 was merged before Jan 19.

From [this doc](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs#select-the-collection-mode):
> Most Azure resources write data to the workspace in either Azure diagnostics or resource-specific mode without giving you a choice. For more information, see [Common and service-specific schemas for Azure resource logs](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema).

> All Azure services eventually use the resource-specific mode. As part of this transition, some resources allow you to select a mode in the diagnostic setting. Specify resource-specific mode for any new diagnostic settings because this mode makes the data easier to manage. It also might help you avoid complex migrations later.

### CODE REVIEW

**Comment by Klaas-** [comment]
@teowa the doc you linked, https://github.com/MicrosoftDocs/azure-docs/blob/e294e228cf8249e82172695143e0a21084636b67/articles/azure-monitor/essentials/resource-logs.md#select-the-collection-mode , it says that we should use Dedicated because AzureDiagnostics will be removed in future, right? So we should maybe put a deprecaion on that and change default in 4.0 or something like that?

**Comment by teowa** [comment]
As I tested, for some target resources (e.g., KeyVault) this property is not settable, and for other resources (e.g., DataFactory) it still works well, see details in https://github.com/Azure/azure-rest-api-specs/issues/22400. I am still contacting with service team about the details of the affected resource types. Temporarily close this and once a solution is determined I am going to reopen this PR.

**Comment by teowa** [comment]
The two cases of  `log_analytics_destination_type` property:
1. for resources (Keyvault), the property is no longer applicable, either set `AzureDiagnostics` or `Dedicated`, returns  null 
2. for others (DataFactory), defaults to `AzureDiagnostics` but can also be changed to `Dedicated`.

We are asking service team to change case1 to return an error. And I think it's better to change this property to `Computed` for now, from the [doc](https://learn.microsoft.com/en-us/azure/azure-monitor/reference/tables/azurediagnostics#azure-diagnostics-mode), case1 is more common and set this as computed can remove the diff noise for users who dont set this property.

File: internal/services/monitor/monitor_diagnostic_setting_resource.go [hunk 1]
```diff
@@ -98,10 +98,10 @@ func resourceMonitorDiagnosticSetting() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
 				ForceNew: false,
-				Default:  "AzureDiagnostics",
```
Comment (katbyte) [inline internal/services/monitor/monitor_diagnostic_setting_resource.go:101]: the documentation will need to be updated?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Review by katbyte — APPROVED** [review]
LGTM 📓

**Pull request merged**